guides/provide-protection.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * This guide covers the following:
+ *   - Verifying that a protection market is valid and supported by the Cozy Protocol
+ *   - Supplying funds to a Protection Market to provide protection
+ */
+
+import hre from 'hardhat';
+import '@nomiclabs/hardhat-ethers';
+import { Contract } from 'ethers';
+import { getChainId, getContractAddress, logSuccess, logFailure, supplyTokensTo, findLog } from '../utils/utils';
+import comptrollerAbi from '../abi/Comptroller.json';
+import cozyTokenAbi from '../abi/CozyToken.json';
+import erc20Abi from '../abi/ERC20.json';
+
+async function main(): Promise<void> {
+  // STEP 0: ENVIRONMENT SETUP
+  // If the signer you are using does not have a balance of over 1 ETH, you'll need to transfer some ETH to it from
+  // this script. Since this is run against a forked network, you can use `hardhat_impersonateAccount` to impersonate
+  // any account with ETH and send it to your signer's address: https://hardhat.org/guides/mainnet-forking.html#impersonating-accounts
+  const supplyAmount = '1000'; // Amount of USDC we want to supply, in dollars (e.g. 1000 = $1000 = 1000 USDC)
+  const provider = hre.ethers.provider;
+  const signer = new hre.ethers.Wallet(process.env.PRIVATE_KEY as string, hre.ethers.provider);
+  const chainId = getChainId(hre);
+  const { MaxUint256 } = hre.ethers.constants;
+  const { parseUnits } = hre.ethers.utils;
+
+  // Since we are testing on a forked Rinkeby and our account has no tokens, we need to initialize the account with
+  // the required tokens
+  await supplyTokensTo('USDC', supplyAmount, signer.address, hre, signer);
+
+  // STEP 1: VERIFY MARKET
+  // We know we'll need the Comptroller, so create an instance the Comptroller contract
+  const comptrollerAddress = getContractAddress('Comptroller', chainId);
+  const comptroller = new Contract(comptrollerAddress, comptrollerAbi, signer); // connect signer for sending transactions
+
+  // The first check is to make sure our protection market is a valid protection market that we can supply to
+  const cUsdcProtectionMarketAddr = '0xbb0e6fC1FE1913DF8f0A483E6dC48329871cb2b7'; // this is the protection market address
```

Comment:
Suggested addition to this comment: "on the Rinkeby testnet"

Reply:
Added in 255c48fa3e36d7e3c85046605dbe41dbd51c2b94